web/app/scripts/views/map/layers-controller.js

```diff
@@ -322,35 +320,38 @@
                 blackspotUrl = TileUrlService.blackspotsUrl(uuid);
                 blackspotUtfUrl = TileUrlService.blackspotsUtfGridUrl(uuid);
             }
+            var secondaryRecordsUrl = $q.when('');
+            var secondaryUtfGridUrl = $q.when('');
+            if (ctl.secondaryType) {
+                secondaryRecordsUrl = TileUrlService.secondaryTilesUrl(ctl.secondaryType);
+                secondaryUtfGridUrl = TileUrlService.recUtfGridTilesUrl(ctl.secondaryType);
+            }
+
             return $q.all(
-                [
-                    TileUrlService.recTilesUrl(ctl.recordType),
-                    TileUrlService.recUtfGridTilesUrl(ctl.recordType),
-                    TileUrlService.recHeatmapUrl(ctl.recordType),
-                    blackspotUrl,
-                    blackspotUtfUrl,
-                    blackspotTileKey
-                ]
+                {
+                    baseRecordsUrl:       TileUrlService.recTilesUrl(ctl.recordType),
+                    baseUtfGridUrl:       TileUrlService.recUtfGridTilesUrl(ctl.recordType),
+                    baseHeatmapUrl:       TileUrlService.recHeatmapUrl(ctl.recordType),
+                    blackspotsUrl:        blackspotUrl,
+                    blackspotsUtfGridUrl: blackspotUtfUrl,
+                    blackspotTileKey:     blackspotTileKey,
+                    secondaryRecordsUrl:  secondaryRecordsUrl,
+                    secondaryUtfGridUrl:  secondaryUtfGridUrl,
+                }
             );
         }
 
         /**
-         * Given an array of urls in form:
-         * [baseRecordsUrl, baseUtfGridUrl, baseHeatmapUrl, blackspotsUrl],
+         * Given the urls object returned by getUrls,
          * updates the layer groups so that the layer switcher does not
          * need to be re-initialized every time layer urls change
          */
         function updateLayerGroups(urls) {
-            var baseRecordsUrl = urls[0];
-            var baseUtfGridUrl = urls[1];
-            var baseHeatmapUrl = urls[2];
-            var blackspotsUrl = urls[3];
-            var blackspotsUtfGridUrl = urls[4];
-            var blackspotTileKey = urls[5];
-
-            updateEventLayer(baseRecordsUrl, baseUtfGridUrl);
-            updateHeatmapLayer(baseHeatmapUrl);
-            updateBlackspotLayer(blackspotsUrl, blackspotsUtfGridUrl, blackspotTileKey);
+            updateEventLayer(urls.baseRecordsUrl, urls.baseUtfGridUrl);
+            updateSecondaryLayer(urls.secondaryRecordsUrl, urls.secondaryUtfGridUrl);
```

Comment:
How are you feeling about the naming scheme here now that you've used it a bunch? There was a thought when Interventions were first set up that they could be kept general, so that they could represent any arbitrary data that the users wanted to track. But reading this is a bit confusing because it's not clear what the Primary is that would necessitate a Secondary.